### ISSUE TITLE
`MutJson` methods `.set()`, `.setAt()`  accept `any`, not `MutJson`

### ISSUE THREAD

**Issue by Chriscbr** [winglang/wing#2767]
### I tried this:

```
bring cloud;

let x = MutJson { a: 1 };
x.set("hello", new cloud.Bucket());
```

### This happened:

No compiler error

### I expected this:

An error saying that `cloud.Bucket` is not of type `Json`

### Component

Compiler

### Wing Version

0.18.7

### Node.js Version

18.14.1

### Platform(s)

MacOS

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by revitalbarletz** [comment]
@staycoolcall911, I changed the method to accept MutJson, when sending any other object, compilation succeed. Tried to convert MutJson to string, then got compilation error. Only with MutJson it seems to be working with every transferred type.

**Comment by staycoolcall911** [comment]
Yes, I think we consider `cloud.Bucket` to be a subtype of `MutJson`.
This also compiles:
```ts
bring cloud;

class A {
    a: MutJson;

    init() {
        this.a = new cloud.Bucket();
    }
}
```
I'll ask Hasan to better understand what we had in mind.

**Comment by hasanaburayyan** [comment]
@staycoolcall911 @revitalbarletz I have some changes in this PR that remove the subtype issue. https://github.com/winglang/wing/pull/3018

**Comment by staycoolcall911** [comment]
This was already fixed by @hasanaburayyan in #3018. Adding a test for it on #3190
